service/opa_transport.go

```diff
@@ -110,16 +108,14 @@ func (t *OPATransport) RoundTrip(req *http.Request) (resp *http.Response, err er
 		return nil, fmt.Errorf("%w: %s", ErrOPATransportInvalidResponseBody, err.Error())
 	}
 
-	userInfo, err := mongoclient.RetrieveUserBindingsAndRoles(rondlogrus.NewEntry(t.logger), t.request, t.userHeaders)
+	pathParams := mux.Vars(t.request)
+	input, err := rondhttp.NewInput(t.request, t.clientHeaderKey, pathParams, t.user, decodedBody)
 	if err != nil {
 		t.responseWithError(resp, err, http.StatusInternalServerError)
 		return resp, nil
 	}
 
-	pathParams := mux.Vars(t.request)
-	input := rondhttp.NewInput(t.request, t.clientHeaderKey, pathParams)
```

Comment:
Here user binding and roles are not refetched, since they are now passed inside the user params. We could think about fetch if user is empty, but at the moment it is not necessary since user is always fetched in input